### ISSUE TITLE
'Make clean' cleans necessary files.

### ISSUE THREAD

**Issue by ja1cul** [01org/QAT_Engine#6]
When I did `make clean`, `e_qat_err.c` and `e_qat_err.h` were purged.
`.tools/Makefile.in` includes command of purging e_qat_err files.

``` diff
diff --git a/.tools/Makefile.in b/.tools/Makefile.in
index 7f021ba..f2c4927 100644
--- a/.tools/Makefile.in
+++ b/.tools/Makefile.in
@@ -801,7 +801,7 @@ uninstall-am: uninstall-includeHEADERS uninstall-libLTLIBRARIES


 mostlyclean-generic:
-       -rm -f *.obj lib tags core .pure .nfs* *.old *.bak fluff *.so *.sl *.dll e_qat_err*
+       -rm -f *.obj lib tags core .pure .nfs* *.old *.bak fluff *.so *.sl *.dll

        -rm -f $(with_openssl_install_dir)/lib/engines-1.1/qat.so
```

Should `Makefile.am` be changed too ?

**Comment by agrandi** [comment]
Hi @ja1cul  

Thanks a lot for your input!
At the moment we are discussing internally how to improve the compilation scripts and make them more consistent.

Our idea is that the `make clean` should delete all the files that are generated in the previous steps, including those created by `make depend`.

However, we are considering to remove the need to run `make depend` entirely so after running `make clean` you would be able to build by running only `make`

Would that work for you?

**Comment by ja1cul** [comment]
Hello @agrandi 

'make depend' before 'make' worked perfect.
Thank you for your advice.

But feel something is wrong.

How do you think an idea below?
> Check existence of nessesary files (e_qat_err.c and so on)
> if exist run target 'depend'
>else run continue.

not so cooll.....

**Comment by agrandi** [comment]
Hi @ja1cul 

Yes, we are considering to add a dependency on the existence of these files.

The logic will be something like:

```python
if e_qat_err.c not exist:
   make depend

make
```

We plan to release these changes in the next release out in about 2 weeks.

**Comment by stevelinsell** [comment]
Actually let me take you through a particular use case to understand how it could be changed.

1) A User wants to enhance the engine by making a change to add a bit of extra functionality. As part of that change they add a new engine specific error message to the code, perhaps in `e_qat.c` for example.
2) When you run `make depend` it runs a script that goes through all the source files (including `e_qat.c` for instance) and generates `e_qat_err.h` and `e_qat_err.c` dynamically by adding the errors it finds in the source files. 
3) If you are rebuilding the project it is not good enough to check for the existence of `e_qat_err.h` and `e_qat_err.c` and not generate them if they already exist as in the case of 1) above the new error message will not get built into the files and you will see a compilation failure.
4) If you run `make clean` there is an expectation that it will clean out files that are generated as part of the build, allowing you to do a fresh compile/link. As the `e_qat_err.h` and `e_qat_err.c` are generated files as part of the build it would be wrong to not clean them on a `make clean` as otherwise running `make` will result in the same failure documented in 3) above.
5) It is not easy to tell whether `e_qat_err.h` and `e_qat_err.c` need regenerating as part of the build. If we want to change to just need to do a `make` and get rid of `make depend` it is safest to always regenerate the files on every `make`.

As such from my point of view we should do the following:
1) Retain removing `e_qat_err.h` and `e_qat_err.c` on a `make clean`.
2) When running a `make` we automatically *always* generate `e_qat_err.h` and `e_qat_err.c` by running the `make depend` step.

Please provide feedback if this does not seem like a sensible way to go.

**Comment by agrandi** [comment]
@stevelinsell this sounds like a good idea!

I am happy to implement your solution :+1:

**Comment by agrandi** [comment]
This has been fixed in https://github.com/01org/QAT_Engine/commit/71c1dbc7412f3e56bfdaf4b956a88be6290dc758 and released in [v0.5.8](https://github.com/01org/QAT_Engine/releases/tag/v0.5.8)
